ntohq/buefy-next
# Create Helper Function for Shallow Fields

Fixes: #168 
```
[Warning] [Vue warn]: Vue received a Component which was made a reactive object. 
This can lead to unnecessary performance overhead, and should be avoided by marking the component with `markRaw` or using `shallowRef` instead of `ref`. 
````

## Proposed Changes

- Remove the warning on every component page located on the Buefy-next [documentation website](https://v3.buefy.org) by:
   - Creating a helper function (`shallowFields`) to shallow ref the components being passed to the Example.vue component.
   - Implement the `shallowFields()` function on every component page that passes a component to the Example.vue component's prop.
   
###  Completion of Changes to All UI Components
- [x] Breadcrumb
- [x] Carousel
- [x] Collapse
- [x] Dialog
- [x] Image
- [x] Loading
- [x] Message
- [x] Navbar
- [x] Progress
- [x] Skeleton
- [x] Sidebar
- [x] Snackbar
- [x] Toast
- [x] Tooltip

**Form Controls**
   - [x] Autocomplete
   - [x] Clockpicker
   - [x] Colorpicker
   - [x] Datepicker
   - [x] Datetimepicker
   - [x] Field
   - [x] Input
   - [x] Numberinput
   - [x] Rate
   - [x] Switch
   - [x] Taginput
   - [x] Timepicker
   - [x] Upload 

### Testing Method
I confirmed that the warning above was no longer present by opening my dev tools, selecting the console tab, and filtering the message by the keyword `reactive`. 



##### Steps to Perform the Test
1. Run the dev server by typing `cd .\packages\docs\` and then `npm run dev` into your console.
2. Open your preferred web browser.
3. Open the dev tools by pressing F12.
4. Select the Console tab.
5. Filter the Console by the keyword `reactive`.
> ![image](https://github.com/ntohq/buefy-next/assets/11604664/4a70db20-2b3a-4636-b41d-c17045962487)
6. Navigate to each component page without closing the dev tools.
7. Check the Console tab to confirm that the warnings are no longer present.
   - If a warning does show, confirm that you completed step 5.
   - If step 5 is still set in the Console tab, please share a screenshot of that component in your review.

### Notes for Reviewing
1. Please confirm that I have passed all components to shallow fields for each Buefy component page.
2. Perform the above test.
